### PR TITLE
Update e2e-tests.md

### LINKED ISSUE #4: Enable e2e tests for Azure
We should set up an e2e tests for this repo

- [x] Setup e2e tests infra on Azure
- [x] Enable e2e tests for PRs
- [x] Switch to test-infra for presubmit e2e tests (presubmit tests are running for pull requests in cloud-provider-azure repo) @ritazh 
  - [x] Add e2e steps to setup test-infra via Makefile
  - [x] Update presubmit tests to the new way
  - [x] Remove current perl scripts
- [ ] Enable full periodic e2e tests for stable releases (periodic tests are running in backgroud for each release braches)
  - [ ] Add more test scenarios, e.g. conformance, correctness, alpha-features, autoscaling, scalability, multi-zone, slow and serial
  - [ ] Enable e2e for more releases, e.g. v1.13 and v1.12
  - [ ] Setup prow jobs for new test scenarios
- [ ] Add Azure features e2e testings #7
  - [ ] LoadBalancer service tests with various annotations
  - [ ] ACR image pulling tests without docker secrets setting explicitly
  - [ ] Multi availability zones tests
  - [ ] Cloud provider configuration options tests (e.g. Standard LoadBalancer and VMSS)
  - [ ] Upgrading tests (e.g. cloud-provider version updates)
- [ ] Documentation for e2e @ritazh

### PR DESCRIPTION
**Pull request by Abhik1998**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
For documentation purposes
**Which issue(s) this PR fixes**:

Fixes #4 

**Special notes for your reviewer**:
Updated single file . Other tests can be updated once resolved

**Release note**:
```
action-required= review
```

### CODE REVIEW

File: docs/e2e-tests.md [hunk 1]
```diff
@@ -64,19 +64,18 @@
...
 4. Deploy a cluster and run smoke test
 
-    Note that 'k8s-azure e2e' command will delete the cluster as last running step. To skip that, add parameter '-cskipcleanup=1' or set environment variable 'K8S_AZURE_SKIPCLEANUP=1'.
-
+    To deploy a cluster:
     ```
-    source <TestProfile>
-    CLUSTER_NAME=<ClusterName>
-    tests/k8s-azure/k8s-azure e2e -cname=$CLUSTER_NAME -cbuild_e2e_test=1 -caccm_image=<username>/azure-cloud-controller-manager:<image_version>
+    make deploy
```
Comment (feiskyer) [inline docs/e2e-tests.md]: some env variables are required before deployment: https://github.com/kubernetes/cloud-provider-azure/blob/master/hack/deploy-cluster.sh#L26

IMAGE_REGISTRY should also be set before running make deploy.

File: docs/e2e-tests.md [hunk 2]
```diff
@@ -64,19 +64,18 @@
...
     ```
 
     To connect the cluster:
     ```
-    source $CLUSTER_NAME/cluster.profile
-    kubectl version
+    cd tests/k8s-azure/manifest
```
Comment (feiskyer) [inline docs/e2e-tests.md]: do we really need to cd?

File: docs/e2e-tests.md [hunk 3]
```diff
@@ -64,19 +64,18 @@
...
+    KUBECONFIG=_output/kubeconfig/kubeconfig.<LOCATION>.json kubectl cluster-info
```
Comment (feiskyer) [inline docs/e2e-tests.md]: export KUBECONFIG is required for e2e

File: docs/e2e-tests.md (hunk 2 above)
Comment (Abhik1998) [reply docs/e2e-tests.md]: Yes sir actually we need to move to the file location else we need to change the deploy-cluster.sh a little more.

File: docs/e2e-tests.md (hunk 1 above)
Comment (Abhik1998) [reply docs/e2e-tests.md]: Sir , it is already there in Step 3.
3. Build a custom image and push it to a testing repository.
    ```
    IMAGE_REGISTRY=<username> make image
    docker push <username>/azure-cloud-controller-manager:<image_version>
    ```

Comment (feiskyer) [reply docs/e2e-tests.md]: yep, but the variable name is different now. We should update them according to latest deploy-cluster.sh

Comment (Abhik1998) [reply docs/e2e-tests.md]: Okay Sir

File: docs/e2e-tests.md [hunk 4]
```diff
@@ -64,19 +64,26 @@
...
-    source <TestProfile>
-    CLUSTER_NAME=<ClusterName>
-    tests/k8s-azure/k8s-azure e2e -cname=$CLUSTER_NAME -cbuild_e2e_test=1 -caccm_image=<username>/azure-cloud-controller-manager:<image_version>
+    #Enter all of the details as in TestProfile
+    RESOURCE_GROUP_NAME=<resource group name>
+    LOCATION=<location>
+    SUBSCRIPTION_ID=<subscription ID>
+    CLIENT_ID=<client id>
+    CLIENT_SECRET=<client secret>
+    TENANT_ID=<tenant id>
```
Comment (feiskyer) [inline docs/e2e-tests.md]: Please move those to step 2 and add export for each

File: tests/k8s-azure/manifest/linux-vmss.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
+{
```
Comment (feiskyer) [inline tests/k8s-azure/manifest/linux-vmss.json]: still not right. the file has already been added in https://github.com/kubernetes/cloud-provider-azure/blob/master/tests/k8s-azure/manifest/linux-vmss.json.

Could you squash all the commits and rebase it to master branch?

File: docs/e2e-tests.md [hunk 6]
```diff
@@ -38,47 +38,39 @@
...
+    kubectl cluster-info
     ```
+    To check out more of the deployed cluster , replace ``` kubectl cluster-info ``` with other `kubectl` commands.
+    To further debug and diagnose cluster problems, use ```kubectl cluster-info dump```
 
-5. Run E2E tests
+4. Run E2E tests
     Please first ensure the kubernetes project locates at `$GOPATH/src/k8s.io/kubernetes`, the e2e tests will be built from that location.
     - Run test suite: default, serial, slow, smoke (smoke suite just tests the cluster is up and do not run any case)
         ```
```
Comment (feiskyer) [inline docs/e2e-tests.md]: This part should also be updated. Please add the scripts for building kubernetes e2e and kubetest

**Pull request closed without merging**